packages/polaris-viz/src/components/DonutChart/components/LegendValues/LegendValues.tsx

```diff
@@ -1,29 +1,73 @@
+/* eslint-disable @shopify/strict-component-boundaries */
 import type {ColorVisionInteractionMethods, DataSeries} from 'index';
-import type {LabelFormatter} from '@shopify/polaris-viz-core';
-import {useTheme} from '@shopify/polaris-viz-core';
+import type {BoundingRect, LabelFormatter} from '@shopify/polaris-viz-core';
+import {
+  COLOR_VISION_SINGLE_ITEM,
+  clamp,
+  getSeriesColors,
+  useChartContext,
+  useTheme,
+} from '@shopify/polaris-viz-core';
+import React, {useRef} from 'react';
 
-import {TrendIndicator} from '../../../TrendIndicator';
 import {getTrendIndicatorData} from '../../../../utilities/getTrendIndicatorData';
-import {SquareColorPreview} from '../../../../components/SquareColorPreview';
+import {HiddenLegendTooltip} from '../../../LegendContainer/components/HiddenLegendTooltip';
+import {useLegend} from '../../../LegendContainer/hooks/useLegend';
+import type {RenderHiddenLegendLabel} from '../../../../types';
 
 import styles from './LegendValues.scss';
+import {useVerticalOverflowLegend} from './utilities/useVerticalOverflowLegend';
+import {LegendValueItem} from './components/LegendValueItem/LegendValueItem';
 
 interface LegendContentProps {
   data: DataSeries[];
+  activeIndex: number;
+  dimensions: BoundingRect;
   labelFormatter: LabelFormatter;
+  renderHiddenLegendLabel?: RenderHiddenLegendLabel;
   getColorVisionStyles: ColorVisionInteractionMethods['getColorVisionStyles'];
   getColorVisionEventAttrs: ColorVisionInteractionMethods['getColorVisionEventAttrs'];
 }
 
 export function LegendValues({
-  data,
+  data: allData,
+  activeIndex,
   labelFormatter,
+  renderHiddenLegendLabel = (count) => `+${count} more`,
   getColorVisionStyles,
   getColorVisionEventAttrs,
+  dimensions,
 }: LegendContentProps) {
   const selectedTheme = useTheme();
+  const {theme} = useChartContext();
 
-  const maxTrendIndicatorWidth = data.reduce((maxWidth, {metadata}) => {
+  const legendItemDimensions = useRef([{width: 0, height: 0}]);
+
+  const seriesCount = clamp({
+    amount: allData.length,
+    min: 1,
+    max: Infinity,
+  });
+
+  const seriesColors = getSeriesColors(seriesCount, selectedTheme);
+
+  const {legend: legendData, height} = useLegend({
+    showLegend: true,
+    data: [{series: allData, shape: 'Bar'}],
+    colors: seriesColors,
+    dimensions,
+  });
+
+  const {displayedData, hiddenData} = useVerticalOverflowLegend({
+    data: legendData,
+    height,
+    legendItemDimensions,
+    enableHideOverflow: true,
+  });
```

Comment:
This component is the _custom_ legend rendered to show values  + trends in the donut chart, using the `overFlowLegend` hook here to get the same overflow behaviour seen in the shared `LegendContainer` 

https://github.com/Shopify/polaris-viz/assets/90475364/3fd68169-787e-4182-ad8c-7cbf4e7c140a